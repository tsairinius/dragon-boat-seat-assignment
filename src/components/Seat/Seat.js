import React from "react";
import Paddler from "../Paddler/Paddler";
import PropTypes from "prop-types";
import styled from "styled-components";

function Seat(props) {
  const paddler = props.paddlersInBoat.filter(
    (paddler) => paddler.seatId === props.id
  );
  console.assert(
    paddler.length <= 1,
    "There should only be at most one paddler assigned to this seat."
  );
  const paddlerComponent = paddler.length ? (
    <Paddler
      paddlerProfile={paddler[0]}
      handlePaddlerMouseEnter={props.handlePaddlerMouseEnter}
      handlePaddlerMouseLeave={props.handlePaddlerMouseLeave}
      handlePaddlerClick={props.handlePaddlerClick}
    />
  ) : undefined;

  return (
    <SeatDiv
      seatId={props.id}
      data-testid={"seat" + props.id}
      onClick={() => props.handleSeatClick(props.id)}
    >
      {paddlerComponent}
    </SeatDiv>
  );
}

Seat.propTypes = {
  id: PropTypes.number.isRequired,
  paddlersInBoat: PropTypes.array.isRequired,
  handleSeatClick: PropTypes.func.isRequired,
  handlePaddlerClick: PropTypes.func.isRequired,
  handlePaddlerMouseEnter: PropTypes.func.isRequired,
  handlePaddlerMouseLeave: PropTypes.func.isRequired,
};

const SeatDiv = styled.div`
  width: 50px;
  height: 50px;
  border: 1px solid #000;
  margin: 0 auto;
  grid-column-start: ${(props) =>
    props.seatId === 21 || props.seatId === 0 ? "1" : "auto"};
  grid-column-end: ${(props) =>
    props.seatId === 21 || props.seatId === 0 ? "3" : "auto"};
  grid-row-start: ${(props) => (props.seatId === 21 ? "22" : "auto")};
  grid-row-end: ${(props) => (props.seatId === 21 ? "23" : "auto")};
`;

export default Seat;
