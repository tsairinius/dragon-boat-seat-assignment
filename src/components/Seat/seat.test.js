import React from "react";
import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Seat from "./Seat";

afterEach(cleanup);

it('renders Seat component without crashing', () => {
    const container = document.createElement('div');
    ReactDOM.render(<Seat />, container);
});

it('rendered Seat matches snapshot', () => {
    const tree = renderer.create(<Seat />).toJSON();
    expect(tree).toMatchSnapshot();
});