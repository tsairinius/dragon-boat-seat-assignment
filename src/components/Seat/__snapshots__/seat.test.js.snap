// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered Seat matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid #000",
      "height": "50px",
      "width": "50px",
    }
  }
/>
`;
